src/Groups/GAPGroups.jl

```diff
@@ -1278,25 +1280,119 @@ false
 """
     is_pgroup(G)
 
+Return `true` if `G` is a ``p``-group for some prime ``p``, that is, if the
+order of every element in `G` is a power of ``p``.
+
+Note that a finite group is a ``p``-group if and only if its order is a prime
+power. In particular, the trivial group is a ``p``-group for every prime.
+
+# Examples
+```jldoctest
+julia> is_pgroup(symmetric_group(1))
+true
+
+julia> is_pgroup(symmetric_group(2))
+true
+
+julia> is_pgroup(symmetric_group(3))
+false
+
+```
+"""
+@gapattribute is_pgroup(G::GAPGroup) = GAP.Globals.IsPGroup(G.X)::Bool
+
+
+"""
+    is_pgroup_with_prime(G)
+
 Return `(true, nothing)` if `G` is the trivial group,
 `(true, p)` if the order of every element in `G` is a power of a prime `p`,
 and `(false, nothing)` otherwise.
 
 For finite groups `G`, the first return value is `true` if and only if
 the order of `G` is a prime power.
+
+# Examples
+```jldoctest
+julia> is_pgroup_with_prime(symmetric_group(1))
+(true, nothing)
+
+julia> is_pgroup_with_prime(symmetric_group(2))
+(true, 2)
+
+julia> is_pgroup_with_prime(symmetric_group(3))
+(false, nothing)
+
+```
 """
-function is_pgroup(G::GAPGroup)
-   if GAPWrap.IsPGroup(G.X)
-      p = GAP.Globals.PrimePGroup(G.X)
-      if p != GAP.Globals.fail
-         return true, fmpz(p)  # TODO: allow specifying the type used for the prime
-      else
-         return true, nothing
-      end
-   end
-   return false, nothing
+function is_pgroup_with_prime(G::GAPGroup)
+  is_trivial(G) && return true, nothing
+  if is_pgroup(G)
+     p = GAP.Globals.PrimePGroup(G.X)
+     if p != GAP.Globals.fail
+        return true, fmpz(p)  # TODO: allow specifying the type used for the prime
+     else
+        return true, nothing
+     end
+  end
+  return false, nothing
 end
 
+"""
+    prime_pgroup(::Type{T} = fmpz, G::GAPGroup) where T <: IntegerUnion
```

Comment:
Perhaps `prime_of_pgroup` would be clearer, and better fit in the overall plan to use self-explanatory function names? Thoughts @fieker @thofma @ThomasBreuer ?

Reply:
The name `prime_of_pgroup` would fit better to `is_pgroup_with_prime`, but `prime_pgroup` is also suitable.
(One could ask why the GAP function in question was not called `PrimeOfPGroup`, since there were already functions such as `GeneratorsOfGroup`.)

Reply:
I am changing this to `prime_of_pgroup`, as I think it is overall clearer (after all someone could in theory introduce "prime $p$-groups"